[Merged by Bors] - chore: rename `dmaSMul_apply` to `domSMul_apply`

YaelDillies opened the pull request:
It's clearer this way, as the `ma` in `dmaSMul` already means "Multiplicative Action", ie `smul`.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

urkud commented:
I see no downsides of this refactor (didn't look at the implementation yet). CC: @eric-wieser , you're better at spotting issues with strange typeclass diamonds etc.

eric-wieser commented:
This effectively defines `DomAct X` as `MulOpposite (AddOpposite X)`, based on the algebraic operations? This feels like it would be pretty surprising.

YaelDillies commented:
Maybe. It's possible that all I want is a nice name for the action by `DomMulAct`/`DomAddAct`. Are we happy to name these `domSMul`/`domVAdd`?

leanprover-community-bot-assistant commented:
This pull request has conflicts, please merge `master` and resolve them.

leanprover-community-bot-assistant commented:
This pull request has conflicts, please merge `master` and resolve them.

leanprover-community-bot-assistant commented:
This pull request has conflicts, please merge `master` and resolve them.

urkud commented:
Please open a Zulip thread and tag Eric and me there.

leanprover-community-bot-assistant commented:
This pull request has conflicts, please merge `master` and resolve them.

YaelDillies committed:
chore: rename `dmaSMul_apply` to `domSMul_apply`
It's clearer this way, as the `ma` in `dmaSMul` already means "Multiplicative Action", ie `smul`.

mattrobball commented:
bors merge
